File: app/src/main/java/android/hitech/com/collapsingtoolbar/MainActivity.java
package android.hitech.com.collapsingtoolbar;

import android.location.Location;

import com.google.android.gms.location.LocationListener;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;

import com.google.android.gms.common.ConnectionResult;
import com.google.android.gms.common.api.GoogleApiClient;
import com.google.android.gms.location.LocationRequest;
import com.google.android.gms.location.LocationServices;

import static android.R.id.primary;

public class MainActivity extends AppCompatActivity implements LocationListener, GoogleApiClient.ConnectionCallbacks,
        GoogleApiClient.OnConnectionFailedListener {
    GoogleApiClient googleApiClient;
    LocationRequest locationRequest;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        locationRequest = new LocationRequest();
        locationRequest.setPriority(LocationRequest.PRIORITY_HIGH_ACCURACY);
        googleApiClient = new GoogleApiClient.Builder(this)
                .addApi(LocationServices.API)
                .addConnectionCallbacks(this)
                .addOnConnectionFailedListener(this)
                .build();
        googleApiClient.connect();
    }

    @Override
    public void onLocationChanged(Location location) {
        Log.i("fusedlocation", String.valueOf(location));
    }

    @Override
    public void onConnected(@Nullable Bundle bundle) {
        LocationServices.FusedLocationApi.requestLocationUpdates(
                googleApiClient, locationRequest, this);
    }

    @Override
    public void onConnectionSuspended(int i) {

    }

    @Override
    public void onConnectionFailed(@NonNull ConnectionResult connectionResult) {

    }
}

File: app/src/main/java/android/hitech/com/collapsingtoolbar/MainActivity.kt
package android.hitech.com.collapsingtoolbar

import android.content.Intent
import android.os.AsyncTask
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import com.google.firebase.database.DatabaseReference
import com.google.firebase.database.FirebaseDatabase
import java.net.URL

class MainActivity : AppCompatActivity(), View.OnClickListener {

    lateinit var userName: EditText
    lateinit var password: EditText
    lateinit var submitBtn: Button
    lateinit var db: FirebaseDatabase
    var user: String? = null
    var pass: String? = null
    lateinit var account: DatabaseReference
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        userName = findViewById(R.id.userName) as EditText
        password = findViewById(R.id.password) as EditText
        submitBtn = findViewById(R.id.submitBtn) as Button
        submitBtn.setOnClickListener(this)

    }

    override fun onClick(v: View) {

    }

    private inner class MyTask : AsyncTask<String, String, String>() {
        override fun doInBackground(vararg params: String?): String {
            val url = URL("")
            val conn = url.openConnection()
            conn.doInput = true
            return conn.getInputStream().bufferedReader().readLine()
        }

        override fun onPostExecute(result: String?) {
            super.onPostExecute(result)
            startActivity(Intent(baseContext, Main2Activity::class.java))
        }
    }
}


